agent/src/protocols/coap_core/discovery_impl.rs

```diff
@@ -0,0 +1,73 @@
+use log::{debug, info};
+
+pub struct CoREResource {
+    uri: String,
+    interface: Option<String>,
+    rtype: Option<String>,
+}
+
+impl CoREResource {
+    fn new(uri: String) -> CoREResource {
+        CoREResource {
+            uri,
+            interface: None,
+            rtype: None,
+        }
+    }
+}
+
+pub fn parse_link_value(link_value: &str) -> Vec<(String, String)> {
+    use coap_lite::link_format::LinkFormatParser;
+
+    let mut parser = LinkFormatParser::new(link_value);
+    let mut resources: Vec<(String, String)> = vec![];
+
+    while let Some(Ok((uri, mut attr_it))) = parser.next() {
+        debug!("Found CoAP resource {}", uri);
+        let mut resource = CoREResource::new(uri.to_string());
+
+        while let Some((attr, value)) = attr_it.next().map(|attr| (attr.0, attr.1.to_string())) {
+            debug!("attr {} value {}", attr, value);
+
+            match attr {
+                "rt" => resource.rtype = Some(value),
+                "if" if value.as_str() == "sensor" => {
+                    // Only "sensor" interface is supported for now
+                    resource.interface = Some(value)
+                }
+                _ => {
+                    // Other attributes are not supported yet
```

Comment:
I should probably panic here

Reply:
we favor passing errors up through the return (rather than panic)

Reply:
My initial comment was wrong. After reading the code again, I think it should be left empty. The rational is that a device can list all of its resources and Akri only picks up the ones it supports. Devices are unaware of Akri and should not forced to limit themselves.